cli.js

```diff
@@ -129,35 +132,45 @@ check({
 
     const runAllTests = !args.extra && !args.missing
 
+    /** @type {string[]} */
+    let extras = []
+    /** @type {string[]} */
+    let result = []
+
     if (runAllTests || args.unused) {
-      const extras = extra(pkg, deps, options)
+      extras = extra(pkg, deps, options)
       failed += extras.length
-      if (extras.length) {
-        console.error('Fail! Modules in package.json not used in code: ' + extras.join(', '))
-      } else if (args.verbose) {
-        console.log('Success! All dependencies in package.json are used in the code')
-      }
     }
     if (runAllTests || args.missing) {
       const optionsForMissingCheck = runAllTests
         ? Object.assign({}, options, {
-            excludeDev: false,
-            excludePeer: false
-          })
+          excludeDev: false,
+          excludePeer: false
+        })
         : options
 
-      const result = missing(pkg, deps, optionsForMissingCheck)
+      result = missing(pkg, deps, optionsForMissingCheck)
 
       failed += result.length
+    }
 
+    if (args.json) {
+      console.log(JSON.stringify({ missing: result, unused: extras }))
+    } else {
+      if (extras.length) {
+        console.error('Fail! Modules in package.json not used in code: ' + extras.join(', '))
+      } else if (args.verbose) {
+        console.log('Success! All dependencies in package.json are used in the code')
+      }
       if (result.length) {
         console.error('Fail! Dependencies not listed in package.json: ' + result.join(', '))
       } else if (args.verbose) {
         console.log('Success! All dependencies used in the code are listed in package.json')
       }
     }
+
     // eslint-disable-next-line promise/always-return
-    process.exit(args.ignore || !failed ? 0 : 1)
+    process.exit(args.json || args.ignore || !failed ? 0 : 1)
```

Comment:
I don't think `--json` should imply `--ignore`

Reply:
You are right. This is not a good idea. I am not sure at the moment why I am adding this.